tt_metal/impl/device/mesh_device.cpp

```diff
@@ -172,6 +178,13 @@ std::shared_ptr<MeshDeviceView> MeshDevice::get_view() {
     return this->view;
 }
 
+std::ostream& operator<<(std::ostream& os, const MeshDevice& mesh_device) {
+    return os << fmt::format("MeshDevice({}x{} grid, {} devices)",
+                             mesh_device.num_rows(),
+                             mesh_device.num_cols(),
+                             mesh_device.num_devices());
+}
```

Comment:
Wouldn't it make more sense for `operator<<` to delegate to `to_string` and not the other way around?

Reply:
yeah, that's better.